gwctl/pkg/resourcediscovery/discoverer.go

```diff
@@ -282,6 +336,29 @@ func (d Discoverer) discoverPolicies(resourceModel *ResourceModel) {
 	resourceModel.addPolicyIfTargetExists(d.PolicyManager.GetPolicies()...)
 }
 
+// discoverEventsForGateways adds Events associated with Gateways that exist in
+// the resourceModel.
+func (d Discoverer) discoverEventsForGateways(ctx context.Context, resourceModel *ResourceModel) {
+	for _, gatewayNode := range resourceModel.Gateways {
+		eventList := &corev1.EventList{}
+		options := &client.ListOptions{
+			FieldSelector: fields.AndSelectors(
```

Comment:
This seems like it would get painful, would it be better to require that Gateway controllers set specific labels on Events they're publishing?

Reply:
Anything specific in your mind about the pain points? I guess some should get addressed as described in the other comment https://github.com/kubernetes-sigs/gateway-api/pull/2999/files#r1585597856 and we may not need it for the time being.